### DISCUSSION TITLE
Using selective persist with multiple slices

### DISCUSSION THREAD

**Discussion by kowaretaken** [pmndrs/zustand#985] [Q&A] [answered]
```
 persist(
    (set, get: GetState<IStore>, api) => ({
      transactionSlice: transactionSlice(set, get, api),
      appSlice: appSlice(set, get, api),
    }),
    {
      partialize: (state) => ({
        transactionSlice: {
          pendingTransactions: Object.fromEntries(
            Object.entries(
              state.transactionSlice.pendingTransactions
            ).filter(([key]) => !['isRunning'].includes(key))
          ),
        },
        appSlice: {
          language: state.appSlice.language,
        },
      }),
    }
  )
```
Using persist middleware,
I want to selectively include certain states within my slices to store, however, on app load, states within slices that are not selected to be persisted would not be initialized. Is there a workaround to this issue?

**Comment by AnatoleLucet** [comment, marked as answer]
This is because the middleware shallow merges the states when hydrating.
Using a deep merge function should help. See https://github.com/pmndrs/zustand/wiki/Persisting-the-store's-data#merge

**Reply by kowaretaken** [reply]
Thank you, using lodash/merge works like a charm!
